### PR TITLE
Peers

### PR DESCRIPTION
**Pull request by tdac**
Add the ability to set peer nodes - allowing the sharing of stick tables

### CODE REVIEW

**Comment by tersmitten** [comment]
Look like a great addition! I would love to merge this.

File: templates/etc/haproxy/peers.cfg.j2 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{% if haproxy_peers is defined %}
+{% for haproxy_peer in haproxy_peers %}
```
Comment (tersmitten) [inline templates/etc/haproxy/peers.cfg.j2]: Can you change this to `{% for haproxy_peer in haproxy_peers | default([]) %}`, this saves an `if`

File: templates/etc/haproxy/peers.cfg.j2 [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+peers {{ haproxy_peer.name }}
+{% for peer in haproxy_peer.peers %}
```
Comment (tersmitten) [inline templates/etc/haproxy/peers.cfg.j2]: Can you change this to `{% for peer in haproxy_peer.peers | default([]) %}`

File: templates/etc/haproxy/haproxy.cfg.j2 [hunk 3]
```diff
@@ -6,6 +6,8 @@ global
 defaults
 {% include 'defaults.cfg.j2' %}
 
+{% include 'peers.cfg.j2' %}
```
Comment (tersmitten) [inline templates/etc/haproxy/haproxy.cfg.j2]: Any reason this is not part of `global`?

File: templates/etc/haproxy/peers.cfg.j2 (hunk 1 above)
Comment (tdac) [inline templates/etc/haproxy/peers.cfg.j2]: amended and checked

File: templates/etc/haproxy/peers.cfg.j2 (hunk 2 above)
Comment (tdac) [inline templates/etc/haproxy/peers.cfg.j2]: amended and checked

File: templates/etc/haproxy/haproxy.cfg.j2 (hunk 3 above)
Comment (tdac) [inline templates/etc/haproxy/haproxy.cfg.j2]: all the examples I've seen have it below 'defaults' but happy to move it if required?

Comment (tersmitten) [inline templates/etc/haproxy/haproxy.cfg.j2]: It is a global setting, right?

Comment (tdac) [inline templates/etc/haproxy/haproxy.cfg.j2]: I can't find anything that conclusively proves it's a global - other than http://cbonte.github.io/haproxy-dconv/configuration-1.5.html#3.5 listing it under "globals" heading.   All the examples I've found couple it with the backends - but I've now moved it to the globals section.

**Pull request merged**